keep/providers/discord_provider/discor_provider.py

```diff
@@ -0,0 +1,84 @@
+"""
+DiscordOutput is a class that implements the BaseOutputProvider interface for Discord messages.
+"""
+import dataclasses
+import pydantic
+import requests
+
+from keep.exceptions.provider_exception import ProviderException
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig
+
+
+@pydantic.dataclasses.dataclass
+class DiscordProviderAuthConfig:
+    """Discord authentication configuration."""
+
+    webhook_url: str = dataclasses.field(
+        metadata={"required": True, "description": "Discord Webhook Url"}
+    )
+
+
+class DiscordProvider(BaseProvider):
+    def __init__(self, provider_id: str, config: ProviderConfig):
+        super().__init__(provider_id, config)
+
+    def validate_config(self):
+        self.authentication_config = DiscordProviderAuthConfig(
+            **self.config.authentication
+        )
+
+    def dispose(self):
+        """
+        No need to dispose of anything, so just do nothing.
+        """
+        pass
+
+    def notify(self, **kwargs: dict):
+        """
+        Notify alert message to Discord using the Discord Incoming Webhook API
+        https://discord.com/developers/docs/resources/webhook
+
+        Args:
+            kwargs (dict): The providers with context
+        """
+        self.logger.debug("Notifying alert message to Discord")
+        webhook_url = self.authentication_config.webhook_url
+        content = kwargs.pop("content", "")
```

Comment:
Perhaps add an exception if both the content and components are empty?

Reply:
added